android/services/mongodb-remote/src/main/java/com/mongodb/stitch/android/services/mongodb/remote/internal/RemoteMongoCollectionImpl.java

```diff
@@ -682,26 +683,54 @@ public ResultT call() {
   }
 
   @Override
-  public Task<ChangeStream<Task<ChangeEvent<DocumentT>>, DocumentT>> watch(final ObjectId... ids) {
+  public Task<ChangeStream<Task<ChangeEvent<DocumentT>>>> watch(final ObjectId... ids) {
     return dispatcher.dispatchTask(
-        new Callable<ChangeStream<Task<ChangeEvent<DocumentT>>, DocumentT>>() {
+        new Callable<ChangeStream<Task<ChangeEvent<DocumentT>>>>() {
         @Override
-        public ChangeStream<Task<ChangeEvent<DocumentT>>, DocumentT> call() throws Exception {
-          return new AsyncChangeStream<DocumentT>(proxy.watch(ids), dispatcher);
+        public ChangeStream<Task<ChangeEvent<DocumentT>>> call() throws Exception {
+          return new AsyncChangeStream<>(proxy.watch(ids), dispatcher);
         }
       }
     );
   }
 
   @Override
-  public Task<ChangeStream<Task<ChangeEvent<DocumentT>>, DocumentT>> watch(final BsonValue... ids) {
+  public Task<ChangeStream<Task<ChangeEvent<DocumentT>>>> watch(final BsonValue... ids) {
     return dispatcher.dispatchTask(
-        new Callable<ChangeStream<Task<ChangeEvent<DocumentT>>, DocumentT>>() {
-        @Override
-        public ChangeStream<Task<ChangeEvent<DocumentT>>, DocumentT> call() throws Exception {
-          return new AsyncChangeStream<DocumentT>(proxy.watch(ids), dispatcher);
+        new Callable<ChangeStream<Task<ChangeEvent<DocumentT>>>>() {
+          @Override
+          public ChangeStream<Task<ChangeEvent<DocumentT>>> call() throws Exception {
+            return new AsyncChangeStream<>(proxy.watch(ids), dispatcher);
+          }
+        }
+    );
+  }
+
+  @Override
+  public Task<ChangeStream<Task<CompactChangeEvent<DocumentT>>>> watchCompact(
+      final ObjectId... ids
+  ) {
+    return dispatcher.dispatchTask(
```

Comment:
[nit] We can do the conversation to BsonObjectIds here and pass them through to the overloaded method. This way, we can collapse the core implementation, and it doesn't have to be aware of the public API, as there is no need for it too.

Reply:
I can do that, but I think the reason it was in core was so that conversion logic only needed to happen in core, and not in both android and server.

Reply:
This was the reason +1